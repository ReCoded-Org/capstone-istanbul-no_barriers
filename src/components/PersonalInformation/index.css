/* Edit profile button styles on click, focus and normal state*/
Button.editProfile.shadow-none.btn.btn-primary,
Button.editProfile.shadow-none.btn.btn-primary:not(:disabled):not(.disabled):active,
Button.editProfile.shadow-none.btn.btn-primary:focus {
  color: var(--main-title-profile);
  border: none;
  background-color: var(--main-text-color-profile);
}

Button.editProfile.shadow-none.btn.btn-primary:hover {
  background-color: var(--blog-background-profile);
  color: var(--main-text-color-profile);
}

.editProfile {
  margin-left: auto;
}

.profilePic {
  height: 10rem;
}

.profileName {
  color: var(--main-title-profile);
}

.bio {
  padding: 0 10rem;
  text-align: justify;
}
