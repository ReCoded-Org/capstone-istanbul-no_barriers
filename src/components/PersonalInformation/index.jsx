import React, { useState } from "react";
import { Row, Col, Button, Image } from "react-bootstrap";
import "./index.css";
import profilePic from "./profile.jpg";
import { useTranslation } from "react-i18next";
import EditProfile from "../EditProfile";

const PersonalInformation = () => {
  const { t } = useTranslation();

  // info state-profile pic should the same from blog
  // name and email should be the same for the user sign Up info
  // bio should be different according to each language
  const [info, setInfo] = useState({
    profilePicSource: profilePic,
    name: "IBO (ﾉ◕ヮ◕)ﾉ*:･ﾟ✧",
    email: "aish@aaa.com",
    phoneNum: "+900000",
    bio:
      "Vestibulum elementum felis magna, sit amet ullamcorper nunc viverra sed. Morbi at est id quam consequat aliquam. Morbi congue tellus quis nisl porta, in varius urna euismod. Aliquam at ligula diam. Fusce accumsan posuere venenatis. Nullam at lacus id diam iaculis finibus nec sed purus. Curabitur eget interdum turpis. Suspendisse pellentesque accumsan mattis. Nullam laoreet dictum metus nec vestibulum. Nulla ac libero a erat vehicula ullamcorper. Phasellus hendrerit mi at leo molestie, nec iaculis tellus suscipit. Vestibulum imperdiet bibendum urna sed condimentum. Sed eu libero nec ex pretium ullamcorper. Vivamus pellentesque leo ipsum, iaculis viverra nunc malesuada ut. Nullam volutpat, ipsum non mollis semper, tortor magna pretium mauris, quis tincidunt arcu lacus vitae diam. Etiam quis augue non libero tristique pellentesque sed quis urna. Proin in porttitor sapien. Class aptent taciti sociosqu ad litora torquent per conubia nostra, per inceptos himenaeos. Quisque porttitor commodo erat nec aliquam. Phasellus hendrerit, urna a congue iaculis, nulla leo posuere nulla, vel accumsan nunc nulla at magna.",
  });
  const [showEditModal, setShowEditModal] = useState(false);

  return (
    <div className="mx-5 px-5">
      <Row>
        <Button
          className="editProfile shadow-none"
          onClick={() => setShowEditModal(true)}
        >
          {t("profile.editProfile")}
        </Button>
        <EditProfile
          setShowEditModal={setShowEditModal}
          showEditModal={showEditModal}
          info={info}
          setInfo={setInfo}
        />
      </Row>
      <Row>
        <Col>
          <div className="d-flex align-items-center justify-content-center mb-5">
            <Image
              src={info.profilePicSource}
              alt="person"
              roundedCircle
              fluid
              className="profilePic mr-5"
            />
            <div>
              <h2 className="profileName">{info.name}</h2>
              <h5 className="pt-2">{t("profile.contactInfo")}</h5>
              <div className="d-flex">
                <p className="pr-3">
                  {t("profile.email")}
                  &nbsp;
                  {info.email}
                </p>
                <p>
                  {t("profile.phone")}
                  &nbsp;
                  {info.phoneNum}
                </p>
              </div>
            </div>
          </div>
        </Col>
      </Row>
      <Row className="bio align-items-center flex-column">
        <h2>{t("profile.bioTitle")}</h2>
        <p>{info.bio}</p>
      </Row>
    </div>
  );
};

export default PersonalInformation;
