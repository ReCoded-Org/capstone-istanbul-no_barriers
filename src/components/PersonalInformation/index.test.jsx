import React from "react";
import renderer from "react-test-renderer";
import PersonalInformation from "./index";

it("PersonalInformation renders correctly", () => {
  const tree = renderer.create(<PersonalInformation />).toJSON();
  expect(tree).toMatchSnapshot();
});
