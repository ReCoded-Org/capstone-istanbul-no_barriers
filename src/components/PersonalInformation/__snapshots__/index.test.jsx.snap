// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PersonalInformation renders correctly 1`] = `
<div
  className="mx-5 px-5"
>
  <div
    className="row"
  >
    <button
      className="editProfile shadow-none btn btn-primary"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      profile.editProfile
    </button>
  </div>
  <div
    className="row"
  >
    <div
      className="col"
    >
      <div
        className="d-flex align-items-center justify-content-center mb-5"
      >
        <img
          alt="person"
          className="profilePic mr-5 img-fluid rounded-circle"
          src="profile.jpg"
        />
        <div>
          <h2
            className="profileName"
          >
            IBO (ﾉ◕ヮ◕)ﾉ*:･ﾟ✧
          </h2>
          <h5
            className="pt-2"
          >
            profile.contactInfo
          </h5>
          <div
            className="d-flex"
          >
            <p
              className="pr-3"
            >
              profile.email
               
              aish@aaa.com
            </p>
            <p>
              profile.phone
               
              +900000
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="bio align-items-center flex-column row"
  >
    <h2>
      profile.bioTitle
    </h2>
    <p>
      Vestibulum elementum felis magna, sit amet ullamcorper nunc viverra sed. Morbi at est id quam consequat aliquam. Morbi congue tellus quis nisl porta, in varius urna euismod. Aliquam at ligula diam. Fusce accumsan posuere venenatis. Nullam at lacus id diam iaculis finibus nec sed purus. Curabitur eget interdum turpis. Suspendisse pellentesque accumsan mattis. Nullam laoreet dictum metus nec vestibulum. Nulla ac libero a erat vehicula ullamcorper. Phasellus hendrerit mi at leo molestie, nec iaculis tellus suscipit. Vestibulum imperdiet bibendum urna sed condimentum. Sed eu libero nec ex pretium ullamcorper. Vivamus pellentesque leo ipsum, iaculis viverra nunc malesuada ut. Nullam volutpat, ipsum non mollis semper, tortor magna pretium mauris, quis tincidunt arcu lacus vitae diam. Etiam quis augue non libero tristique pellentesque sed quis urna. Proin in porttitor sapien. Class aptent taciti sociosqu ad litora torquent per conubia nostra, per inceptos himenaeos. Quisque porttitor commodo erat nec aliquam. Phasellus hendrerit, urna a congue iaculis, nulla leo posuere nulla, vel accumsan nunc nulla at magna.
    </p>
  </div>
</div>
`;
